### PR TITLE
Add sonarqube pack

### PR DESCRIPTION
**Pull request by ferhatvurucu**

### CODE REVIEW

**Comment by RickyGrassmuck** [comment]
I used to have the pleasure of maintaining a Sonarqube deployment on Nomad and wanted to point out that Sonarqube requires the system it runs on to have a couple of `sysctl` values set to levels that are higher than the system defaults(at least for Ubuntu 20.04 and Centos 7). Just looked at our client servers configuration and we had to have the following `sysctl` values set in order for the application to startup.

```
vm.max_map_count=262144
fs.file-max=65536
```

It's been a while since we stopped using Sonarqube so this may have changed but it's likely worth adding a note/warning in the README about the possibility of needing to change system level settings on the clients the job is going to be scheduled on.

**Comment by ferhatvurucu** [comment]
> I used to have the pleasure of maintaining a Sonarqube deployment on Nomad and wanted to point out that Sonarqube requires the system it runs on to have a couple of `sysctl` values set to levels that are higher than the system defaults(at least for Ubuntu 20.04 and Centos 7). Just looked at our client servers configuration and we had to have the following `sysctl` values set in order for the application to startup.
> 
> ```
> vm.max_map_count=262144
> fs.file-max=65536
> ```
> 
> It's been a while since we stopped using Sonarqube so this may have changed but it's likely worth adding a note/warning in the README about the possibility of needing to change system level settings on the clients the job is going to be scheduled on.

I updated the client requirements. Thanks a lot for the contribution.

File: packs/sonarqube/README.md [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+```hcl
+client {
+  host_volume "sonarqube" {
+    path      = "/opt/sonarqube/data"
+    read_only = false
+  }
+}
+```
+
+- If you're running on Linux, you must ensure to set these values by running the following commands as root:
```
Comment (jrasell) [inline packs/sonarqube/README.md:29]: Would you be able to expand on why we need these values so future readers can get context on why they are needed?

File: packs/sonarqube/variables.hcl [hunk 2]
```diff
@@ -0,0 +1,97 @@
+variable "job_name" {
+  description = "The name to use as the job name which overrides using the pack name."
+  type        = string
+  default = "sonarqube"
```
Comment (jrasell) [inline packs/sonarqube/variables.hcl:4]:
```suggestion
  description = "The name to use as the job name which overrides using the pack name."
  type        = string
  default     = "sonarqube"
```

File: packs/sonarqube/templates/sonarqube.nomad.tpl [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      read_only = false
+      source    = "[[.sonarqube.volume_name]]"
+    }
+    [[- end ]]
+
+    restart {
+      attempts = 2
+      interval = "5m"
+      delay = "15s"
+      mode = "fail"
```
Comment (jrasell) [inline packs/sonarqube/templates/sonarqube.nomad.tpl:61]:
```suggestion
      attempts = 2
      interval = "5m"
      delay    = "15s"
      mode     = "fail"
```

**Review by jrasell — CHANGES_REQUESTED** [review]
Hi @ferhatvurucu, this looks great and works really well thanks! I have a couple of minor comments that I would like to get addressed before merging.

**Comment by ferhatvurucu** [comment]
Hi @jrasell,

Thanks a lot for your review and contribution. I updated the pack based on your comments.

**Review by jrasell — APPROVED** [review]
LGTM, thanks @ferhatvurucu!

**Pull request merged**